app/controllers/topics/bookmarks_controller.rb

```diff
@@ -0,0 +1,55 @@
+class Topics::BookmarksController < ApplicationController
+  def show
+  end
+
+  def new
+    #@topic = Topic.find(params[:id])
+    @bookmark = Bookmark.new
+  end
+
+  def edit
```

Comment:
your action here just needs to load the bookmark and topic. no need to call save